packages/glimmer-runtime/lib/compiled/blocks.ts

```diff
@@ -61,6 +61,14 @@ export class InlineBlock extends Block {
     let ops = new InlineBlockCompiler(this, env).compile();
     return this.compiled = new CompiledBlock(ops, this.symbolTable.size);
   }
+
+  initBlocks(blocks = this['children'], parentTable = this['symbolTable']): this {
```

Comment:
I don't think you should overload `InlineBlock`; instead, make a new top level `PartialBlock` that does the set of things you're trying to do here.